types/types.go

```diff
@@ -1001,3 +1001,15 @@ func IsSelectorsInTags(tags, selectors []string) bool {
 
 	return true
 }
+
+func GetKubernetesNodeProviderFrom(providerURL string) string {
```

Comment:
Can we add cluster info related function to one file instead? This one seems being not used in other places. 

Also, can we change the name to `GetKubernetesProviderNameFromURL` instead?

https://github.com/longhorn/longhorn-manager/blob/b260399001bcf7bd40796eabf91889fbc17ff003/vendor/k8s.io/kubernetes/pkg/apis/core/types.go#L4378-L4378

Reply:
> Can we add cluster info related function to one file instead? This one seems being not used in other places.
> 

To be honest, I am not sure where to move this to.
I think it will be better to reorganize the common packages as a whole.

https://github.com/longhorn/longhorn/issues/6039

> Also, can we change the name to `GetKubernetesProviderNameFromURL` instead?
> 

Updated.

Reply:
Sounds good